tools/debug/debugsymbolviewer.py

```diff
@@ -0,0 +1,177 @@
+ram_text_start_addr = 0x000000
+ram_text_end_addr = 0x000000
+flash_text_start_addr = 0x000000
+flash_text_end_addr = 0x000000
+
+# Function to setup the Address to Symbol mapping table from the System Map file
+def setup_symbol_table(sm_file):
+	global flash_text_start_addr
+	global flash_text_end_addr
+	global ram_text_start_addr
+	global ram_text_end_addr
+	is_ram_text = False
+	# Reading the System Map file and preparing the symbol map table
+	with open(sm_file) as searchfile:
+		for line in searchfile:
+			s = line.split(' ', 2)
+			if len(s) == 3:
+				if not '$' in s[2]:
+					# s[0] contains address and s[2] contains Symbol name
+					symbol_lookup_table.append((int(s[0], 16), s[2].rstrip()))
+
+					# Extract kernel text start & end addresses from the table
+					symbol = s[2].rstrip('\n')
+					if symbol == '_stext':
+						flash_text_start_addr = hex(int(s[0], 16))
+					if symbol == '_etext':
+						flash_text_end_addr= hex(int(s[0], 16))
+					if symbol == '_ram_kernel_text_start':
+						is_ram_text = True
+						ram_text_start_addr = hex(int(s[0], 16))
+					if symbol == '_ram_kernel_text_end':
+						ram_text_end_addr = hex(int(s[0], 16))
+
+	if debug:
+		# Printing the Created Symbol table
+		print('~~~~~~~~~~~~~~~~~~~~~~~~ SYMBOL TABLE START ~~~~~~~~~~~~~~~~~~~~~')
+		for line in symbol_lookup_table:
+			print("{0:x} {1}".format(line[0], line[1]))
+		print('~~~~~~~~~~~~~~~~~~~~~~~~SYMBOL TABLE END   ~~~~~~~~~~~~~~~~~~~~~')
+
+		# Printing the kernel text address ranges in ram and flash areas
+		print('~~~~~~~~~~~~~~~~~~~~~~~~~~  KERNEL TEXT RANGE ~~~~~~~~~~~~~~~~~~~~~~~')
+		print('flash_text_start_addr\t=\t', flash_text_start_addr)
+		print('flash_text_end_addr\t=\t', flash_text_end_addr)
+		if (is_ram_text):
+			print('ram_text_start_addr\t=\t', ram_text_start_addr)
+			print('ram_text_end_addr\t=\t', ram_text_end_addr)
+		print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
+		print('')
+
+
+# Function to search the search_addr present in the System map file and print its corresponding symbol
+def print_symbol(search_addr):
+	addr = 0x0
+	next_addr = 0x0
+	l = 0
+	h = len(symbol_lookup_table)
+	mid = (l + h) / 2
+
+	while (l <= h):
+		addr = symbol_lookup_table[mid - 1][0]
+		next_addr = symbol_lookup_table[mid][0]
+
+		if (search_addr >= addr and search_addr <= next_addr):
+			print("Address:  {1}\tSymbol:  {0}".format(symbol_lookup_table[mid - 1][1], hex(addr)))
+			break
+
+		if (search_addr < addr):
+			h = mid - 1
+		else:
+			l = mid + 1
+
+		mid = (l + h) / 2
+
+	if (l > h):
+		print('Symbol not found in system map file')
+
+
+# Function to check if address lies in the kernel text address range
+def is_kernel_text_address(address):
+
+	# Check the address
+	if (address >= ram_text_start_addr and address < ram_text_end_addr) or (address >= flash_text_start_addr and address < flash_text_end_addr):
+		if debug:
+			print("{0} is kernel text address!".format(address))
```

Comment:
Instead of printing this line here, we can leave it. 
While trying to print the symbol, we will either print the address and corresponding symbol. Or we print that the symbol for this address is not found.